spec/support/api_spec_helper.rb

```diff
@@ -287,20 +282,26 @@ def expect_single_resource_query(attr_hash = {})
 
   def expect_single_action_result(options = {})
     expect(response).to have_http_status(:ok)
-    expect(response_hash).to include("success" => options[:success]) if options.key?(:success)
-    expect(response_hash).to include("message" => a_string_matching(options[:message])) if options[:message]
-    expect(response_hash).to include("href" => a_string_matching(fetch_value(options[:href]))) if options[:href]
-    expect_result_to_represent_task(response_hash) if options[:task]
+    expected = {}
+    expected["success"] = options[:success] if options.key?(:success)
+    expected["message"] = a_string_matching(options[:message]) if options[:message]
+    expected["href"] = a_string_matching(fetch_value(options[:href])) if options[:href]
+    if options[:task]
+      expected["task_id"] = anything
+      expected["task_href"] = anything
+    end
+    expect(response_hash).to include(expected)
   end
 
   def expect_multiple_action_result(count, options = {})
     expect(response).to have_http_status(:ok)
-    expect(response_hash).to have_key("results")
-    results = response_hash["results"]
-    expect(results.size).to eq(count)
-    expect(results.all? { |r| r["success"] }).to be_truthy
-
-    results.each { |r| expect_result_to_represent_task(r) } if options[:task]
+    expected_result = {"success" => true}
+    if options[:task]
+      expected_result["task_id"] = anything
+      expected_result["task_href"] = anything
```

Comment:
moving away from DRY.  task_id and task_href that we check for should be defined once, method, constant, etc.